### PR TITLE
Global texture wrap

### PR DESCRIPTION
**Pull request by aferriss**
Building on top of https://github.com/processing/p5.js/pull/2960 this pr implements a top level textureWrap function. textureWrap changes the default wrap mode for all textures in a sketch. This should close https://github.com/processing/p5.js/issues/2189

This change also adds a textureWrapX & textureWrapY property to the p5.RendererGL in order to keep track of the wrap state.

### CODE REVIEW

File: src/webgl/material.js [hunk 1]
```diff
@@ -298,6 +298,73 @@ p5.prototype.texture = function(tex) {
...
+ * is a power of two size (128, 256, 512, 1024, etc.).
+ *
+ * This method will affect all textures in your sketch until a subsequent
+ * textureWrap call is made.
+ *
+ * If only one argument is provided, it will be applied to both the
+ * horizontal and vertical axes.
+ * @method textureWrap
+ * @param {Constant} wrapX Required, either CLAMP, REPEAT, or MIRROR
+ * @param {Constant} wrapY Optional, either CLAMP, REPEAT, or MIRROR
```
[the remark this answers is not in the document]
Comment (aferriss) [reply src/webgl/material.js]: Ah did not know that, thanks!

**Comment by aferriss** [comment]
Hey I'm not sure what to do about this travis build error. It's saying that the MIRROR constant doesn't exist when it goes to do the linting of the samples. It was added in https://github.com/processing/p5.js/pull/2960 though...

**Comment by kjhollen** [comment]
I'll take a look at this in a little bit. From what I've seen, this usually means there's some other kind of error somewhere... @AidanNelson saw something similar last week with p5.Camera stuff.

**Comment by Spongman** [comment]
sometimes travis just fails randomly, i have seen similar issues before. try closing & reopening the PR - that will cause travis to trigger another build.

**Comment by AidanNelson** [comment]
I ran into a similar issue a few days ago where the `no-undef` error turned out to be caused by a totally unrelated error in my YUI documentation (I hadn't specified a type with an `@param` tag).  I also had a problem with `@for` tags, but that shouldn't be an issue here because everything is added to the p5.prototype.

Maybe [this](https://github.com/aferriss/p5.js/blob/6a30b2f1441dc944dce3b3e237d2d05195632fdc/src/webgl/material.js#L475) is the culprit?
```
 * @return {Number[]]}  Normalized numbers array
```

**Comment by kjhollen** [comment]
good find @AidanNelson! I gave this a try, but it didn't get rid of the error (though we should still fix it!)

I had to add this before the MIRROR declaration in core/constants.js:

```
  /**
   * @property {String} MIRROR
   * @final
   */
  MIRROR: 'mirror',
```

It looks like the example linter wants all referenced constants to have documentation.

re: Travis build errors, there are intermittent test failures that cause build failures from time to time, though this wasn't one of them. if you check the logs and find that this has happened, you can log in to Travis with your github account and kick off a new build as Stalgia suggests (or I'll do it when I review) :)

**Comment by kjhollen** [comment]
examples look great, tests are great. one quick additional request that I have: could you add descriptions for what happens when you pick CLAMP, MIRROR, or REPEAT in the docs?

**Comment by stalgiag** [comment]
Hi @aferriss I know you are probably busy but in the interest in keeping your work from dying on the vine I updated your branch and made the requested changes [here.](https://github.com/stalgiag/p5.js/tree/aferrisGlobalTextureWrapPR3047) Not sure the best way to do this but if you are okay with it I'll just merge through another PR referencing this one.

**Comment by stalgiag** [comment]
closing as work was grabbed and updated in #3537. Thanks @aferriss !

**Comment by aferriss** [comment]
@stalgiag Thanks for much for keeping this alive! 🎊

**Pull request closed without merging**